ts/Data/DataTableCore.ts

```diff
@@ -120,7 +119,7 @@ class DataTableCore {
 
     public readonly id: string;
 
-    public modified: DataTableCore;
+    public modified?: this;
```

Comment:
The same problem as above: https://github.com/highcharts/highcharts/pull/23365/files#r2239076738